### ISSUE TITLE
MySQL via the VIP broken in master

### ISSUE THREAD

**Issue by cbaenziger** [bloomberg/chef-bach#1274]
Trying to build a Chef-BACH master cluster fails right now with
```
Recipe: bcpc::powerdns
  * chef_vault_secret[mysql-pdns] action nothing (skipped due to action :nothing)
  * mysql_database[pdns] action create

    ================================================================================
    Error executing action `create` on resource 'mysql_database[pdns]'
    ================================================================================

    Mysql2::Error::ConnectionError
    ------------------------------
    Can't connect to MySQL server on '10.0.109.5' (111)
```

It seems the problem is that HAProxy is configured for MySQL via the following (I'm using 109 as my subnet not 100):
```
listen mysql-galera 192.168.109.5:3306
  timeout  client 24h
  timeout  server 24h
  mode tcp
  balance leastconn
  option  tcplog
  option tcpka
  option httpchk
  server f-iptables-bcpc-vm1.bcpc.example.com 192.168.109.11:3306 check port 3307 inter 1s rise 1 fall 1 observe layer4 backup
```

Meanwhile, MySQL is configured via [my.cnf](https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc/templates/default/mysql/my.cnf.erb#L19) to be using the management network (which are the 10 addresses).

The issue seems to be in #1269, the values of `server['ip']` were used([1](https://github.com/bloomberg/chef-bach/pull/1269/files#diff-67b7e2ac498a57db37e331dd5c4aaa7bR61),[2](https://github.com/bloomberg/chef-bach/pull/1269/files#diff-67b7e2ac498a57db37e331dd5c4aaa7bR74)) rather than the proper service IP's.

**Comment by leochen4891** [comment]
yes, I realized that `node[bach_web][links]` and other two hashes (`files` and `service_ports`) should be set to empty hash. I was using a wrapper during the test which had hidden this issue. Thanks for catching it.

**Comment by vt0r** [comment]
Fixed in #1275
